### PR TITLE
[UI, NG] Select Option Component

### PR DESCRIPTION
**Pull request by adityarb88**
PR Scope:

1. Create `clr-option` component.
2. Style `clr-option` component.
3. Tests

Tested on Chrome

Signed-off-by: Aditya Bhandari <adityab@vmware.com>

### CODE REVIEW

File: src/clr-angular/forms/select/option.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+@Component({selector: "clr-option", templateUrl: "option.html", host: {"[class.clr-option]": "true"}})
+export class ClrOption {
+    constructor(private ifOpenService: IfOpenService, @Optional() @Inject(POPOVER_HOST_ANCHOR) parentHost: ElementRef) {
+        if (!parentHost) {
+            throw new Error("clr-option should only be used inside of a clr-select");
+        }
+    }
+
+    @HostListener("click")
+    closeMenuOnClick() {
```
Comment (youdz) [inline src/clr-angular/forms/select/option.ts]: Can you just add a short comment saying this "close on click" behavior should _not_ be kept for multi-selects? I'm just afraid we might forget. Thanks.

File: src/clr-angular/forms/select/option.spec.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+class TestComponent {
+}
+
+export default function(): void {
+    describe("Select Option Component", function() {
+        let context: TestContext<ClrOption, TestComponent>;
+        let ifOpenService: IfOpenService;
+
+        beforeEach(function() {
+            context = this.create(ClrOption, TestComponent, [IfOpenService]);
```
Comment (Shijir) [inline src/clr-angular/forms/select/option.spec.ts]: Do we need `IfOpenService` provided here again? I see it's provided at the component level above?

File: src/clr-angular/forms/select/option.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+ */
+import {Component, ElementRef, HostListener, Inject, Optional} from "@angular/core";
+import {POPOVER_HOST_ANCHOR} from "../../popover/common/popover-host-anchor.token";
+import {IfOpenService} from "../../utils/conditional/if-open.service";
+
+@Component({selector: "clr-option", templateUrl: "option.html", host: {"[class.clr-option]": "true"}})
+export class ClrOption {
+    constructor(private ifOpenService: IfOpenService, @Optional() @Inject(POPOVER_HOST_ANCHOR) parentHost: ElementRef) {
+        if (!parentHost) {
+            throw new Error("clr-option should only be used inside of a clr-select");
```
Comment (Shijir) [inline src/clr-angular/forms/select/option.ts]: I usually test if the expected error is being thrown or not if the `parentHost` is missing.

File: src/clr-angular/forms/select/option.spec.ts (hunk 2 above)
Comment (adityarb88) [reply src/clr-angular/forms/select/option.spec.ts]: Dont think we do. Will remove it. Thanks!

File: src/clr-angular/forms/select/option.ts (hunk 3 above)
Comment (adityarb88) [reply src/clr-angular/forms/select/option.ts]: 😠. (:trollface: will test that.)

File: src/clr-angular/forms/select/option.ts (hunk 1 above)
Comment (adityarb88) [reply src/clr-angular/forms/select/option.ts]: Yep. Forgot to note that.

**Pull request merged**